### ISSUE TITLE
[BUG] Types break when getting all documents in a collection without query

### ISSUE THREAD

**Issue by KDanisme** [tylim88/FirelordJS#94]
1. Version: latest
2. Meta Type =
`{
  title: string;
  type: 'object';
  properties: {
    [name: string]: 'string' | 'boolean' | 'integer';
  };
};
`
4. Trying to run `(await getDocs(this.db.collection())).docs`
5. What I expect to get: 
When using `(await getDocs(query(this.db.collection))).docs`
The docs getter is of type: `get QuerySnapshot<MetaTypeCreator<Schema, "schema", string, null, { allFieldsPossiblyReadAsUndefined: false; banNull: false; }>>.docs(): QueryDocumentSnapshot<MetaTypeCreator<Schema, "schema", string, null, {allFieldsPossiblyReadAsUndefined: false, banNull: false}>>[]`
6. What Im Actually Getting:
The type of docs getter is:
 `get QuerySnapshot<MetaType>.docs(): QueryDocumentSnapshot<MetaType>[]`

TLDR: 
`getDocs(query(this.db.collection()))` works
`getDocs(this.db.collection())` doesn't
Even tho docs state it should: https://firebase.google.com/docs/firestore/query-data/get-data#get_all_documents_in_a_collection

**Comment by tylim88** [comment]
one of your base data type field is `{ [name: string]: 'string' | 'boolean' | 'integer';   }; }`

as for now I do not support mapped type, this is my reason:

![image](https://user-images.githubusercontent.com/5227509/219948890-cebfef5b-689f-43b4-b09a-c24d56a43a17.png)

This is something I am considering for V3, my decision for now is, not to implement it as it goes against Firestore best practise

I may change my mind if you can provide me good reasons

I will see what I can do with `getDocs(this.db.collection())`

**Comment by tylim88** [comment]
I am sorry, I will continue tomorrow as I am very tired now

**Comment by tylim88** [comment]
fix at [2.3.6](https://github.com/tylim88/FirelordJS/releases/tag/2.3.6)

please try it
